app/serializers/current_user_serializer.rb

```diff
@@ -325,4 +326,24 @@ def include_has_topic_draft?
   def draft_count
     object.user_stat.draft_count
   end
+
+  def include_recent_searches?
+    SiteSetting.log_search_queries
+  end
+
+  def recent_searches
```

Comment:
If the site setting is enabled this is an extra query per logged in user. That might be OK, but I wonder if we should consider only querying this if they use "search"? The downside is of course that search would show up slower.

Reply:
Yeah, it's a trade-off, but I'm leaning more towards what you're suggesting, i.e. making a request when invoking search. I guess we'll see if the delay in displaying it is a nuisance.